### DISCUSSION TITLE
Centos 8.3 first launch after fresh install ==> HTTP error 500

### DISCUSSION THREAD

**Discussion by grassel82** [monarc-project/MonarcAppFO#335] [Q&A] [answered]
Hi,

we are in the process of installing Monarc at our company to be used by the DPO and myself. Our company policy requires Red Hat 8 and no Debian/Ubuntu. As the OVA file published is a little old from the specs (Ubuntu 18.04), I tried to implement everything on Centos 8 before pushing the configuration to Red Hat 8 virtual machine.

I followed the guide from https://www.konzeptacht.de/monarc-installation-ubuntu-18-04 with the option 20.04. I tried the guide out, install on ubuntu-20.04 worked flawlessy. I took the same steps, adapted a few, but then when connecting the first time to the main page, i encounter a strange HTTP error 500.

After debugging and analysing log files from PHP, in the apache log files were no errors, the following error is shown:
`[28-Feb-2021 23:17:28 Europe/Luxembourg] PHP Fatal error:  Uncaught ProxyManager\Exception\FileNotWritableException: Could not move file "/tmp/temporaryProxyManagerFile7WbrC0" to location "/var/www/html/monarc/fo/data/LazyServices/Proxy/ProxyManagerGeneratedProxy__PM__MonarcCoreModelTableUserTokenTableGenerated645f96ba34a1ea00b6312cacf4de5b99.php": either the source file is not readable, or the destination is not writable in /var/www/html/monarc/fo/vendor/ocramius/proxy-manager/src/ProxyManager/Exception/FileNotWritableException.php:19
Stack trace:
#0 /var/www/html/monarc/fo/vendor/ocramius/proxy-manager/src/ProxyManager/GeneratorStrategy/FileWriterGeneratorStrategy.php(85): ProxyManager\Exception\FileNotWritableException::fromInvalidMoveOperation()
#1 /var/www/html/monarc/fo/vendor/ocramius/proxy-manager/src/ProxyManager/GeneratorStrategy/FileWriterGeneratorStrategy.php(58): ProxyManager\GeneratorStrategy\FileWriterGeneratorStrategy->writeFile()
#2 /var/www/html/monarc/fo/vendor/ocramius/proxy-manager/src/ProxyManager/Factory/AbstractBaseFactory.php(110): ProxyManag in /var/www/html/monarc/fo/vendor/laminas/laminas-servicemanager/src/ServiceManager.php on line 772`

So he is not able to write a temporary file in /tmp/... folder or copy an existing file from there to "ProxyManager" folder in Data directory.

I have disabled SELINUX to be sure that no blocking is on the /tmp folder for processes.

Setup:
`Centos 8.3: Linux localhost.localdomain 4.18.0-240.10.1.el8_3.x86_64 #1 SMP Mon Jan 18 17:05:51 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux` 
`PHP: 7.4.15 (cli) (built: Feb  2 2021 14:19:57) ( NTS )`
`mariadb-server.x86_64: 3:10.3.27-3.module_el8.3.0+599+c587b2e7`
`nodejs: v14.16.0`
`composer: 2.0.11`

Does somebody has a solution to this problem?

Kind regards,

Gilles Rassel
City of Luxembourg
Luxembourg

**Comment by K8Sudo** [comment]
Hey @grassel82,

did you create the `data/LazyServices/Proxy` directory?
I just did a test install on Centos 8.3 and there were no problems. During the week I will try to release a tutorial for Centos 8.3 on our website.

Kind regards,

Marc Sparwel

**Comment by ruslanbaidan** [comment]
Hello @grassel82,

@KonzeptAcht , Thanks for your comment.

In additional, you need also to validate if your /var/www/html/monarc/fo/data directory is writable for the www-data user.

**Comment by K8Sudo** [comment]
@grassel82 Here is the new installation guide for Centos 8. This way the installation works fine for me.
[https://www.konzeptacht.de/monarc-installation-centos-8](https://www.konzeptacht.de/monarc-installation-centos-8)
If you have any comments or problems please write again

Kind regards,

Marc Sparwel

**Comment by grassel82** [comment]
Hi,

Thak you for the updated procedure. I followed your recommendations from yesterday with changing write access to data folder and monarc web app worked. I had some issues with selinux, I investigating to fix them. There was also no possibility to set a password on new user accounts. I will try to create a new test vm tomorrow and to follow up your procedure.

I will keep you updated.

Kind regards,

Gilles

**Reply by ruslanbaidan** [reply]
Hello,

There is a possibility to set a password for a new user in the UI and the password should have minimum length of 9 characters and must contain at least: one uppercase, one number and one special character.
We need to see why error message is not displayed in the UI.

**Reply by ruslanbaidan** [reply]
The issue is fixed and can be deployed today:
https://github.com/monarc-project/MonarcAppFO/issues/337

**Comment by grassel82** [comment, marked as answer]
Good evening,

I deployed an updated version today and it works.

Thank you very much for your help and support.

I will install the production server tomorrow, the documentation worked well and there were no issues I could see for the moment.

Kind regards

Gilles
